server/src/rooms/lobby/tournament/state.ts

```diff
@@ -12,49 +9,51 @@ const logger = settings.logging.getLogger(__filename);
 export class TournamentLobbyRoomState extends LobbyRoomState {
   maxClients = 200;
 
-  clientsToGroups = new Map<string, LobbyClient[]>();
+  // only holds clients that still need to be put into a group
+  queue: LobbyClient[] = [];
+  // groups indexed by a unique group id
+  pendingGroups: Map<string, LobbyClient[]> = new Map();
+  // map client username to group id
+  clientToGroup: Map<string, string> = new Map();
```

Comment:
Would a `Group` class with a fixed `size`, `LobbyClient[]` and `id` simplify things? Then `clientToGroup = Map<string, Group>`

Reply:
Almost certainly, could also just store the group reference on the LobbyClient which I think was how the old tournament lobby did it but that'll either be a case of a really messy extension of the LobbyClient schema class or putting it in the base class to be a dead property elsewhere. The map is probably fine.. only needed for one operation